spec/form_models/publishers/job_listing/applying_for_the_job_form_spec.rb

```diff
@@ -38,14 +38,36 @@
     end
 
     context "when the current organisation given is not a local authority" do
-      let(:organisation) { build_stubbed(:trust) }
-      let(:vacancy) { build_stubbed(:vacancy, job_roles: %w[teacher]) }
+      subject { described_class.new(current_organisation: organisation, vacancy: vacancy) }
 
-      it "does not override enable_job_applications" do
-        subject.valid?
+      context "when the vacancy is in draft" do
+        let(:vacancy) { build_stubbed(:vacancy, :draft, job_roles: %w[teacher]) }
+
+        it "validates enable_job_applications and how_to_apply" do
+          subject.valid?
+
+          expect(subject.errors).to include(:enable_job_applications)
+        end
+
+        it "does not override enable_job_applications" do
+          subject.enable_job_applications = true
+
+          expect { subject.valid? }.to_not(change { subject.enable_job_applications })
+        end
+      end
+
+      context "when the vacancy has been listed and enable_job_applications is nil" do
+        let(:vacancy) { build_stubbed(:vacancy, :past_publish, job_roles: %w[teacher], enable_job_applications: nil) }
+
+        it "overrides enable_job_applications" do
+          expect { subject.valid? }.to change { subject.enable_job_applications }.from(nil).to(false)
+        end
+
+        it "does not validate enable_job_applications and how_to_apply" do
+          subject.valid?
 
-        expect(subject.enable_job_applications).to be_nil
-        expect(subject.errors).to include(:enable_job_applications)
+          expect(subject.errors).to_not include(:enable_job_applications)
```

Comment:
and how_to_apply

Reply:
So either change the test description to exclude how_to_apply, or test for it

Reply:
It would be good to test that is was valid, too? `expect(subject).to be_valid`

Reply:
It's not valid though as it does not provide values for the other fields. Should I provide values for the other fields, or should I just test for the absence of the error related to `enable_job_applications` ?